Write data analysis chapter (#76)
daffidwilde/thesis: chapters/data/main.tex

\chapter{An exploratory analysis of administrative data}

\renewcommand{\texpath}{chapters/data/paper/tex}

This chapter provides a exploratory analysis of a patient-episode dataset
provided by the Cwm Taf Morgannwg University Health Board (UHB). This dataset
details, among other administrative quantities, the costs associated with
treating patients during their time in hospital.

The purpose of this analysis is to locate any surface-level sources of variation
in these costs. In particular, this analysis considers a selection of attributes
associated with costs, their distributions across the whole dataset, and how
they interact with one another. These attributes are comprised of non-trivial
cost components and a set of clinical attributes that are known to drive costs.

The subsequent analysis reveals that, while the bulk of the data corresponds to
short-stay and relatively low-impact spells of treatment, there are long, heavy
tails with high levels of variation in each of these variables. As such, a more
homogeneous part of the population should be considered to find more actionable
results.

> In aid of this, a strategy for the analysis of slices within the data is

Author: I didn't really like framework but you're right. Approach is better-suited. Fixed by 
4dc415a

established, using the diabetic population as an example. This framework
provides another dimension to the overall analysis through the use of comparison
and contrast, but the intended impact is ultimately lost due, again, to high
levels of variation.

The chapter is structured as follows:
\begin{itemize}
    \item Section~\ref{sec:overview} provides an overview of the dataset and
        its key attributes
    \item Section~\ref{sec:diabetes} explores the subset of the data
        corresponding to diabetic patients
    \item Section~\ref{sec:summary} summarises the findings of this analysis
\end{itemize}

\section{An overview of the data}\label{sec:overview}
\graphicspath{{chapters/data/paper/img/external/}}

\subsection{Data structure}\label{subsec:structure}

> Before any analysis can be conducted, the structure of the data must be

Author: Fixed by 71559e8

understood, as well as how it has been prepared. This dataset comprises

> approximately two and a half million episode records for patients from across

Author: I believe it would be "is comprised of" but that's in the passive voice, so I've opted for "comprises".

Wales that were treated in the Prince Charles and Royal Glamorgan hospitals
(South Wales) from April 2012 through April 2017. An approximation for the
geographic distribution of patients in the dataset is given in
Figure~\ref{fig:proportion_wales}, where it is clear that the patient population
are mostly from the local area.

\begin{figure}
    \centering
    \includegraphics[width=\imgwidth]{proportion_wales.pdf}
    \caption{%
        The proportion of patients observed in the dataset by postcode district
        (e.g.\ CF24)%
    }\label{fig:proportion_wales}
\end{figure}

An episode is defined to be any continuous period of care provided by the same
consultant in the same place~\cite{NHS:episode}. For instance, if a patient is

> admitted to a general medical ward for diagnosis and testing, and then is

Author: Good catch, yes.

> admitted to a general medical ward for diagnosis and testing, and then is

Author: Fixed by 71559e8

referred to a specialist consultant in oncology, then their first episode would
end with their testing, and a second episode of care would begin on the oncology
ward. Each of these episodes would correspond to a row in the dataset. If the
patient was then immediately discharged, they would have completed a spell with
two episodes.

Looking at the episodes directly will be avoided in this analysis. Instead, this
analysis favours aggregating a patient's episodes into spells. Statistics
associated with these aggregates are referred to as \emph{spell-level}
statistics. The reason for this level of aggregation is that, since the

> introduction of the `payment by results' system for financial flows, it has been

Author: Fixed by 71559e8

any given patient.

\subsection{Cleaning the data}\label{subsec:formatting}

As with any data analysis, a substantial amount of preprocessing and formatting
is required to make the data sufficiently consistent and suitable for analysing.
With the dataset at hand, this process included the removal of some superfluous
attributes which added unwanted redundancy to the dataset, and a number of rows
that had been corrupted or coded incorrectly. In addition to this, some columns
have been reformatted; namely those whose entries were intended to be used as
date-time objects such as admission and discharge dates.

It has already been stated that the majority of the attributes in the dataset
will not be considered in this analysis. By ignoring these attributes, the focus
is purely on how the costs of care appear in the data. The subset of chosen
attributes will frequently be referred to as the set of \emph{key attributes}
but this choice of name does not imply that the remaining attributes are not of
interest nor that they are in any way unimportant.

The key attributes provide a base for understanding how the costs and resources

> consumed by a patient in a spell originate: cost components give direct

Author: That does appear to be confusing. Is it not clear that episodes being aggregated into their spells?

displayed in Figures~\ref{fig:no_spells}~---~\ref{fig:netcost}, it is clear
that the data is weighted towards low-cost, short-stay, and otherwise low-impact
patients. This behaviour is especially clear in
Figures~\ref{fig:no_spells}~and~\ref{fig:los}. Here, it is clear that, of all
the spells provided under the care of the health board, the majority are
day-cases. Also, the patients being treated are one-time users of the hospital
system.

\begin{figure}
    \centering
    \includegraphics[width=\imgwidth]{nspells_bar.pdf}
    \caption{Number of spells associated with each patient}%
    \label{fig:no_spells}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\imgwidth]{los_bar.pdf}
    \caption{Bar chart for length of stay}%
    \label{fig:los}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\imgwidth]{max_diags.pdf}
    \caption{Maximum number of diagnoses in each spell}%
    \label{fig:no_diag}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\imgwidth]{no_procs.pdf}
    \caption{Total number of procedures in each spell}%
    \label{fig:no_proc}
\end{figure}

> \begin{figure}

Author: I haven't found any particular benefit from using it so I figured things were cleaner this way.

If there was a case where I absolutely needed to have a figure exactly somewhere then I'd either give it its own page with the `\afterpage` environment (from the `afterpage` package) or use the `[H]` locator from the `float` package.

    \centering
    \includegraphics[width=\imgwidth]{netcost.pdf}
    \caption{Kernel density estimate for the net cost of a spell}%
    \label{fig:netcost}
\end{figure}

> In general, the distributions themselves have long, pronounced tails, suggesting

Author: There's no need for it really given that the tails are there and we're talking about them. I'll remove it. 👍 

```suggestion
In general, the distributions themselves have long tails, suggesting
```

grouping ages together to aid the calculation of risk factors and projected
costs. This approach has proven to be particularly helpful when looking at older
patients~\cite{Billings327}, but is limited in scope as will be discussed in
Section~\ref{sec:diabetes}. Baring this in mind, however, studying the
distribution of age among the patients can provide another valuable insight into
how costs may appear.

\begin{figure}
    \centering
    \includegraphics[width=\imgwidth]{age.pdf}
    \caption{%
        Age of patients in the dataset compared with the estimated UK population
        in 2016
    }\label{fig:age}
\end{figure}

Figure~\ref{fig:age} shows this distribution in contrast to a UK population
estimate in 2016 from the Office for National Statistics (ONS). Following the
graph from left to right, the UK estimate is roughly uniform from birth up until
the late fifties where a decline appears as older people become less prevalent.
Looking instead at the distribution belonging to the patients, it is clear that
there are several peaks and troughs. The largest trough corresponds to
adolescents which makes sense anecdotally since some of the least likely people

> to visit a hospital would be reaching their peak fitness biologically.

Author: That certainly feels like a more considerate wording. Will add.

Similarly, the distinct peaks around infancy and in the older age range
often correspond to those people who are most vulnerable in terms of their
health. Thus, a hospital should expect to see a disproportionate number of
people at those ages.

So, by looking at these key attributes, it would appear things are as expected:
people tend to go to their local hospitals, and historically vulnerable people
are more likely to go. However, there is significant spread in the costs,
severity and lengths of hospital visits. Moreover, the likelihood of returning
to hospital seems relatively low for the vast majority of the population served
by the health board. All of these things are straightforward. However, they do
not make this a fruitless exercise as getting to grips with any body of data is
absolutely essential. In fact, the analysis thus far has shown that this
population is typical, in some (broadly anecdotal) respects, of many other

> populations.

Author: Fixed by 49781b0

\subsection{Pairwise correlation}\label{subsec:corr}

Looking at the univariate distributions of the key attributes in the previous
section gave a good base for understanding the scope of the data. As such, the
next logical step is to investigate how these key attributes interact with one
another. In this analysis, correlation coefficients will be used to give a sense
of this interaction.

Figure~\ref{fig:corr_heatmap} shows the Pearson correlation coefficient between
all the pairs of key attributes. These correlation coefficients have been
presented in the form of a heat map with a colour bar, indicating the scale of
the correlation between any two variables. Using a visualisation such as this is
more intuitive than reading directly from an array of numbers, and makes gaining

> insight from the relationships between variables easier. The attributes

Author: I agree. I skimmed over this on my way through, and it is a relic from the analysis we gave to CTMUHB way back when.

themselves have been arranged into descending order according to their summed
absolute correlation coefficient. This reordering makes it easier to deduce
which variables have the most prominent levels of interaction.

\begin{definition}
    Consider a dataset with \(m \in \mathbb{N}\) columns,
    \(A = \left\{A_1, \ldots, A_m\right\}\). Attribute \(A_j\) has
    associated with it a \emph{summed absolute correlation coefficient},
    \(c_j\), given by:
    \begin{equation}\label{eq:abs_corr}
        c_j = \sum_{k=1}^{m} \left\| \rho_{A_j, A_k} \right\|
    \end{equation}

    Here, \(\rho_{A_j, A_k}\) is the Pearson's correlation coefficient
    between attributes \(A_j\) and \(A_k\).
\end{definition}

\begin{figure}
    \resizebox{\textwidth}{!}{%
        \includegraphics[width=\linewidth]{corr_heatmap.pdf}
    }
    \caption{%
        Pairwise correlation coefficients for the key cost attributes
    }\label{fig:corr_heatmap}
\end{figure}

Upon inspection of the heat map, there are many cost components that have no
substantial linear correlation with any of the other attributes. The absence of

> correlation here only adds to the evidence that the patients present in the data
> present themselves to hospital with a wide array of needs. Having said that,

Reviewer: ```suggestion
correlation here only adds to the evidence that the patients in the data
present themselves to hospital with a wide array of needs. Having said that,
```

there are clear correlations between several of the attributes; some of these
are easier to realise than others.

For instance, ignoring the main diagonal, the largest value is that between

> total costs (COST) and net costs (NetCost) with a value of 0.94. This high value

Reviewer: Do you want to put variable names like these in a different font? Perhaps `\texttt{COST}` ?

Author: That's a good idea. I'll go through and pick those up.

Author: Fixed by bec4e1d

costs is to be expected.

Other examples of strong correlation are those between length of stay
(TRUE\_LOS) and ward and overhead costs (WARD and OVH respectively). These are
well-known relationships that can be justified succinctly: the longer a patient
spends in hospital, the more time they are likely to spend on a ward. Thus,
incurring associated overheads like administrative work, cleaning costs and a
larger proportion of rental costs. It should also be clear that these three
attributes all share a strong linear correlation with the net cost of a spell,
suggesting that these costs and the length of stay are strong indicators of the
net cost of treating someone, and may suggest that the remaining cost components
make up a substantially smaller part of the net cost.

\subsection{Measuring variation and importance in our cost components}

The broader purpose of this chapter is to better understand the factors leading
to variation in the cost of treating patients. Therefore, it would be fitting to
investigate how this variation can be attributed to each of the cost components.
By doing so, a high-level indication of which departments and procedures that

> impose more (or less) variation can be identified. Once a level of variation has

Reviewer: Do they impose variation or do they "create variation"?

Author: Yeah, that's a good point. I think when I originally wrote this I was thinking about this from the perspective of a patient spell where it could appear that variation is imposed upon the spell.

That feels pretty clumsy now that I think about it, and misses the point of the analysis as something to do with the system. I'll change that now.

```suggestion
create more (or less) variation can be identified. Once a level of variation has
```

of this non-scaling is evident in the standard deviations of
Table~\ref{tab:summary}.

\begin{definition}
    Consider a population with mean \(\mu\) and standard deviation \(\sigma\).
    Then the \emph{coefficient of variation}, denoted by \(C_v\), is defined to
    be:
    \begin{equation}\label{eq:coeff_var}
        C_{v} := \frac{\sigma}{\mu}
    \end{equation}

    If only a sample of the data from a population is available then the
    coefficient of variation can be estimated using the sample standard
    deviation and the sample mean.
\end{definition}

Figure~\ref{fig:cost_variation} shows the coefficient of variation for each of
the cost components. The components have been ranked as in
Figure~\ref{fig:corr_heatmap} from the most to least correlated. It is
immediately clear that there are a number of highly variant cost components.
Take outpatient costs (OUTP) as an example: its standard deviation is over
thirty times the size of its mean. This relative heterogeneity could go some way
in explaining why there seemed to be no linear correlation with the other
variables in Figure~\ref{fig:corr_heatmap}.

> At the other end of the figure, ward and overhead costs have some of the

Reviewer: ```suggestion
At the other end of Figure~\ref{fig:corr_heatmap}, ward and overhead costs have some of the
```

\begin{figure}[h]
    \centering
    \includegraphics[width=\imgwidth]{cost_contribution.pdf}
    \caption{%
        Average contribution of each cost component to the net cost of a spell
    }\label{fig:cost_contribution}
\end{figure}

By inspecting Figure~\ref{fig:cost_contribution}, it is seen that ward, overhead
and medical (MED) costs are the largest contributors to the net cost of a spell
by a significant margin. When looking across the remaining bars, the
contribution is substantially smaller for the department-specific cost
components. Not only that but it appears that the most varied components (from
Figure~\ref{fig:cost_variation}) have near negligible average contributions to
the net cost of a spell.

So the question left to be answered is: can these small but highly varied
components be considered especially important? And what about the other
components? The midriffs of each of these figures contain many of the same
components but the relationships are less clear. In order to visualise how these
two quantities relate to one another, a bubble plot is used. Such a plot allows

> for three-dimensional data to be displayed in the two-dimensional plane; by

Reviewer: In contrast to my comment about the visualisation of the correlation coefficients, this paragraph doesn't niggle at me at all. You're explaining the specific purpose of the chosen visualisation.

markers are also of interest since these components indicate that the level of
variation is relatively low there, perhaps indicating the component has been
optimised somehow.

This figure clearly indicates that the conclusions made previously still hold:
that the largest contributors have some of the smallest measures of
variation. Meanwhile, the smallest average contributors are more strongly
varied. What is of interest is the jump between these components and the others.
There does not seem to be any particular component in the midriff of
contributors that has large, or small, variation. As such, a deeper
investigation is required to properly analyse individual components and
their relationships with specific types of patient.

\section{Diabetic patient analysis}\label{sec:diabetes}
\graphicspath{{chapters/data/paper/img/diabetes/}}

The main conclusion to be taken away from the previous analysis was that the
dataset contains a significant amount of variation. Therefore, in order to
conduct more meaningful analysis, more homogeneous subsets of the data must be
considered.

Classically, patients are categorised by age or condition. However, doing so

> often gives an unrepresentative slice of patients~\cite{Vuik2016a}. In this

Reviewer: non representative?

(I don't know.)

Author: I prefer unrepresentative to non-representative, personally. Would uncharacteristic be better?

patients represent a population whose spells are more severe on average than the
typical patient. Therefore, they will likely have a larger effect on the
hospital system on the whole. Again, a more detailed breakdown of the skeleton
for each of these attributes as well as the other key attributes is given in
Table~\ref{tab:diab_summary}. This table also shows a comparison between both
populations being considered in this section.

\afterpage{%
    \clearpage%
    \thispagestyle{empty}
    \begin{landscape}
        \begin{table}
        \centering
            \resizebox{.8\paperheight}{!}{%
                \input{\texpath/diab_summary.tex}
            }
            \caption{%
                Spell-level statistics for each of the key attributes in the
                diabetic population (and non-diabetic population in parentheses)
            }\label{tab:diab_summary}
        \end{table}
    \end{landscape}
}

\begin{figure}
    \centering
    \includegraphics[width=\imgwidth]{age.pdf}
    \caption{%
        Bar chart for the age of patients in the presence of diabetes and not
    }\label{fig:diab_age}
\end{figure}

The patient age distribution for each group is given in

> Figure~\ref{fig:diab_age}. This figure quite clearly shows how unrepresentative

Reviewer: and here `unrepresentative` reads completely right to me... 🤷‍♂️

    }\label{fig:diab_corr_difference}
\end{figure}

To more directly see the subtleties between these correlation coefficients and
those of the general population (shown in Figure~\ref{fig:corr_heatmap}),
another heat map has been included to show their differences in
Figure~\ref{fig:diab_corr_difference}. This heat map utilises a different colour
map to reflect this, and the attributes have been ranked in descending order of
their summed absolute differences.

This figure indicates that drug and therapy costs (DRUG and THER respectively)
have the largest total difference in correlation coefficients. In fact, the
signs of each coefficient are the same, and so we can say that the diabetic
population has more strongly correlated drug and therapy costs. The same
pattern occurs with a number of other cost components, including the core
components from the general population: ward, medical and overhead costs. These
increased correlation coefficients may be indicative of some intrinsic property
the diabetic population holds. However, they could merely be a coincidence of
considering a subset of the dataset that is, by default, more homogeneous.

> Other than the few attributes at the top, this difference heat map shows that

Reviewer: ```suggestion
Other than the few attributes at the top, this heat map shows that
```

optimised for both the diabetic and non-diabetic populations. This
`optimisation' is only up to the point where the smallest relative variation of
a component is still twice its mean.

\subsection{Resource consumption}\label{subsec:diab_resources}

The types of comparisons made between the non-diabetic and diabetic populations
throughout this analysis are useful for observing their similarities in a direct
way, and in understanding how the groups may relate to one another. However,
these are not the only devices available for examining such a subset of the
data. Particularly when looking at costing data such as this, another useful way
of evaluating a subset is to quantify its contributions to those costs over time
with regard to the general population. The attributes used to identify these
contributions can give a sense of the level and nature of the resources that are
consumed by the population in question. This stage of the analysis considers the
following attributes: the proportion of net costs and admissions, and the length
of stay. 

For these purposes, the data must be manipulated into a chronological form.

> Therefore, some approximations must be made. Here, each of the chosen attributes

Reviewer: Are these approximations or just the specific way we're choosing to define these things?

Author: This is true. The explanation after this (lines 776-783) covers the issues that I have called "approximations" here. I'll get rid of that phrase now.

```suggestion
Here, each of the chosen attributes
```

evolve over the entire data period. In each case, the weekly, monthly and yearly
means are shown. The data has been aggregated in this way, rather than using the
daily data, in an attempt to smooth out the misrepresentation that is described
above. In addition to these plotted points, the data has been fitted with a
standard least-squares linear regression model.

Figures~\ref{fig:admissions}~and~\ref{fig:netcost_proportions} suggest that the
amount of resources consumed by the diabetic population is increasing slowly.
The former indicates that, on average, the number of diabetic patients
visiting the hospital is increasing at a rate of approximately one percent over
five years. From the latter, it is seen that the yearly average proportion of
net spending on diabetic patients has also experienced a shallow increase of
roughly half a percent over the same period.

In addition to this, both figures show some form of divergence over time, as
shown by the spread in the weekly and monthly averages. This is an interesting
phenomenon; there seems no apparent reason for this variability to increase in
recent years with improved policy on prevention, diagnosis, management and

> treatment~\cite{NHS:ltp,NICE,Penn2018,PHE}. The droops toward the end of the

Reviewer: ```suggestion
treatment~\cite{NHS:ltp,NICE,Penn2018,PHE}. The drops toward the end of the
```

Can you double check how your spell checker missed this @daffidwilde ?

Reviewer: > Can you double check how your spell checker missed this @daffidwilde ?

lol - just realised that droops is a word...

Author: Are you okay with droop?

Reviewer: That's a good point actually, I think drop is better :+1:

Author: Fixed by 89c7a11

        regression model.}%
    \label{fig:los_time}
\end{figure}

Across each of the models displayed here, there appears to be some seasonality.
The handling of seasonal behaviour in a regression model has more to do more
with the semantics of finding a ``good'' regression model than was intended here
but it is an important concept nonetheless. If the purpose of this exercise was
to accurately predict the quantities being plotted rather than just seeing the
general trend, then a more elaborate model should have been fitted.

\section{Chapter summary}\label{sec:summary}

This chapter provided a descriptive analysis of an administrative dataset
provided by the Cwm Taf Morgannwg UHB. This dataset describes the episodes of
all patients being treated at two of their hospitals over a five year period.
The scope of this exploration of the dataset was thwarted by the span and
diversity of the episodes therein. In fact, there was little insight to be
gained beyond its conforming to some loosely anecdotal concept of a
administrative healthcare dataset.

Following this initial analysis, a similar methodology was employed on a slice

> of the dataset. As an act of familiarity, the diabetic population was used. The

Reviewer: ```suggestion
of the dataset. Following advice from the subject matter expert, the diabetic population was used. The
```

What do you reckon?

Author: Much more diplomatic 👍

hope of that analysis was that taking a slice according to some characteristic
would result in a more homogeneous dataset, which was confirmed, although only

> marginally. Other than this, there were very few actionable insights to be

Reviewer: ```suggestion
marginally. Other than this, there were few actionable insights to be
```

Author: Fixed by 4dc415a

gained from this sort of exploratory analysis.

> So what was there to gain by looking at the diabetic population? The lack of

Reviewer: ```suggestion
So what was there to gained by looking at the diabetic population? The lack of
```

Author: ```suggestion
So what was there to be gained by looking at the diabetic population? The lack of
```

novel insights may have been expected since the decision to look at diabetic
patients was effectively arbitrary. Moreover, the decision framework was not
descriptive enough to indicate that any particular kind of patient was being
investigated other than that they must exhibit this one condition. So, in that
way, there was little to gain. However, as has been noted throughout this
chapter, taking a subset of the population allows for some comparison with its
complement, as well as the general population.

The next chapter considers another administrative dataset comprised of patients
presenting another chronic condition. The work in that chapter incorporates some
of the techniques introduced in the previous chapters, but it also builds on the
surface-level analyses employed here. The most important of these analyses is
the slicing of a dataset to identify more homogeneous parts, and then comparing
those. The process by which this is done incorporates more attributes of the

> data than the condition alone, or age, as is commonly done. Ultimately, the

Reviewer: Is it worth here, connecting this chapter to the other chapters? Perhaps just indicating that in the next chapter we will use techniques discussed in the previous chapters?

Author: Isn't that covered by lines 861-864?

Reviewer: I did somehow miss that (🤦 ) but I think it could be even more explicit. 

```
The work in that chapter will rely on / use ... presented in chapter ...
```

Author: Fixed by 89c7a11 (maybe?)
